pkg/auth/handler/webapp/viewmodels/preview_authflow_branch.go

```diff
@@ -0,0 +1,487 @@
+		FlowType:           authflow.FlowTypeLogin,
+		ActionType:         authflow.FlowActionType(config.AuthenticationFlowStepTypeAuthenticate),
+		DeviceTokenEnabled: false,
+		Branches:           branches,
+	}
+}
+
+func (m *InlinePreviewAuthflowBranchViewModeler) NewAuthflowBranchViewModelForInlinePreviewUsePasskey() AuthflowBranchViewModel {
+	loginIDKeyType := m.getFirstLoginIDKeyType()
+	branches := m.generateAuthflowBranchesIdentityLoginID(loginIDKeyType)
+	branches = slice.Filter[AuthflowBranch](branches, func(b AuthflowBranch) bool {
+		return b.Authentication != config.AuthenticationFlowAuthenticationPrimaryPasskey
+	})
+	return AuthflowBranchViewModel{
+		FlowType:           authflow.FlowTypeLogin,
+		ActionType:         authflow.FlowActionType(config.AuthenticationFlowStepTypeAuthenticate),
+		DeviceTokenEnabled: false,
+		Branches:           branches,
+	}
+}
+
+func (m *InlinePreviewAuthflowBranchViewModeler) NewAuthflowBranchViewModelForInlinePreviewEnterTOTP() AuthflowBranchViewModel {
+	loginIDKeyType := m.getFirstLoginIDKeyType()
+	branches := m.generateAuthflowBranchesLoginIDAuthenticateSecondary(loginIDKeyType)
+	branches = slice.Filter[AuthflowBranch](branches, func(b AuthflowBranch) bool {
+		return b.Authentication != config.AuthenticationFlowAuthenticationSecondaryTOTP
+	})
+	return AuthflowBranchViewModel{
+		FlowType:           authflow.FlowTypeLogin,
+		ActionType:         authflow.FlowActionType(config.AuthenticationFlowStepTypeAuthenticate),
+		DeviceTokenEnabled: false,
+		Branches:           branches,
+	}
+}
+
+func (m *InlinePreviewAuthflowBranchViewModeler) NewAuthflowBranchViewModelForInlinePreviewOOBOTPLink() AuthflowBranchViewModel {
+	branches := m.generateAuthflowBranchesIdentityLoginID(model.LoginIDKeyTypeEmail)
+	branches = slice.Filter[AuthflowBranch](branches, func(b AuthflowBranch) bool {
+		return b.Authentication != config.AuthenticationFlowAuthenticationPrimaryOOBOTPEmail
+	})
+	return AuthflowBranchViewModel{
+		FlowType:           authflow.FlowTypeLogin,
+		ActionType:         authflow.FlowActionType(config.AuthenticationFlowStepTypeAuthenticate),
+		DeviceTokenEnabled: false,
+		Branches:           branches,
+	}
+}
+
+func (m *InlinePreviewAuthflowBranchViewModeler) NewAuthflowBranchViewModelForInlinePreviewCreatePassword() AuthflowBranchViewModel {
+	loginIDKeyType := m.getFirstLoginIDKeyType()
+	branches := m.generateSignupFlowBranchesIdentityLoginID(loginIDKeyType)
+	branches = slice.Filter[AuthflowBranch](branches, func(b AuthflowBranch) bool {
+		return b.Authentication != config.AuthenticationFlowAuthenticationPrimaryPassword
+	})
+	return AuthflowBranchViewModel{
+		FlowType:           authflow.FlowTypeSignup,
+		ActionType:         authflow.FlowActionType(config.AuthenticationFlowStepTypeCreateAuthenticator),
+		DeviceTokenEnabled: false,
+		Branches:           branches,
+	}
+}
+
+func (m *InlinePreviewAuthflowBranchViewModeler) getFirstLoginIDKeyType() model.LoginIDKeyType {
```

Comment:
so is it a terminology problem, i.e. change the method name to `mostAllowedAuthnMethod` or logic problem?

Reply:
@tung2744

Reply:
A logic problem. For example, in case of primary authentication. If your first login id is username, the current logic will only display `password` and `passkey`. But if you also enabled phone number as one of the login id, you may also want to preview `oob_otp` branch, even phone number is not the first login id.

Therefore, I think the logic should be the "largest available branches".

Reply:
As per offline discussion, let's ask @fungc-io on this in meeting